src/v/storage/disk_log_impl.cc

```diff
@@ -334,15 +334,39 @@ ss::future<> disk_log_impl::do_compact(compaction_config cfg) {
       "[{}] applying 'compaction' log cleanup policy with config: {}",
       config().ntp(),
       cfg);
-    // find first not compacted segment
+
+    // create a logging predicate for offsets..
+    auto offsets_compactible = [&](segment& s) {
+        if (log::has_compactible_offsets(s, cfg)) {
+            vlog(
+              gclog.debug,
+              "[{}] segment {} stable offs {}, max compactible {} -> compact.",
```

Comment:
nit: would be great if you could change this message to be more descriptive. now the `->` doesn't really mean anything without having a context Would be better if it says:
```
compacting segment
```

Reply:
`->` is mathematical notation for "implies". I can change it to "compacting" but "segment" is redundant IMHO

Reply:
Changed to "compacting" in latest update.. thanks for the suggestion.